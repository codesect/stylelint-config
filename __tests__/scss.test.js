const fs = require('fs');
const stylelint = require('stylelint');

const config = require('../scss');
const invalidScss = fs.readFileSync('./__tests__/invalid.scss', 'utf-8');
const validScss = fs.readFileSync('./__tests__/valid.scss', 'utf-8');

describe('Valid CSS config', () => {
  let result = null;

  beforeEach(async () => {
    result = await stylelint.lint({
      code: validScss,
      config,
      syntax: 'scss',
    });
  });

  it('does not error if CSS is valid', () => {
    expect(result.errored).toBeFalsy();
  });

  it('raises no warnings if CSS is valid', () => {
    expect(result.results[0].warnings.length).toBe(0);
  });
});

describe('Invalid SCSS config', () => {
  let result = null;

  beforeEach(async () => {
    result = await stylelint.lint({
      code: invalidScss,
      config,
      syntax: 'scss',
    });
  });

  it('throws error if SCSS is invalid', () => {
    expect(result.errored).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(result.results[0].warnings).toMatchSnapshot();
  });
});
