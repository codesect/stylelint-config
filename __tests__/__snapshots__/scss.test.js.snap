// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid SCSS config matches snapshot 1`] = `
Array [
  Object {
    "column": 1,
    "line": 14,
    "rule": "scss/at-function-parentheses-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before parentheses in function declaration (scss/at-function-parentheses-space-before)",
  },
  Object {
    "column": 1,
    "line": 19,
    "rule": "scss/at-import-no-partial-leading-underscore",
    "severity": "error",
    "text": "Unexpected leading underscore in imported partial name (scss/at-import-no-partial-leading-underscore)",
  },
  Object {
    "column": 15,
    "line": 22,
    "rule": "scss/at-import-partial-extension",
    "severity": "error",
    "text": "Unexpected extension \\".scss\\" in @import (scss/at-import-partial-extension)",
  },
  Object {
    "column": 3,
    "line": 26,
    "rule": "scss/at-mixin-argumentless-call-parentheses",
    "severity": "error",
    "text": "Expected parentheses in mixin \\"test\\" call (scss/at-mixin-argumentless-call-parentheses)",
  },
  Object {
    "column": 3,
    "line": 31,
    "rule": "scss/at-mixin-named-arguments",
    "severity": "error",
    "text": "Expected a named parameter to be used in at-include call (scss/at-mixin-named-arguments)",
  },
  Object {
    "column": 3,
    "line": 31,
    "rule": "scss/at-mixin-named-arguments",
    "severity": "error",
    "text": "Expected a named parameter to be used in at-include call (scss/at-mixin-named-arguments)",
  },
  Object {
    "column": 1,
    "line": 35,
    "rule": "scss/at-mixin-parentheses-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before parentheses in mixin declaration (scss/at-mixin-parentheses-space-before)",
  },
  Object {
    "column": 1,
    "line": 45,
    "rule": "scss/at-rule-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown at-rule \\"@unknown\\" (at-rule-no-unknown) (scss/at-rule-no-unknown)",
  },
  Object {
    "column": 9,
    "line": 62,
    "rule": "scss/dollar-variable-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" (scss/dollar-variable-colon-space-after)",
  },
  Object {
    "column": 14,
    "line": 65,
    "rule": "scss/dollar-variable-colon-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\":\\" (scss/dollar-variable-colon-space-before)",
  },
  Object {
    "column": 3,
    "line": 71,
    "rule": "scss/dollar-variable-no-missing-interpolation",
    "severity": "error",
    "text": "Expected variable $counter to be interpolated when using it with counter-reset (scss/dollar-variable-no-missing-interpolation)",
  },
  Object {
    "column": 1,
    "line": 78,
    "rule": "scss/double-slash-comment-empty-line-before",
    "severity": "error",
    "text": "Expected empty line before comment (scss/double-slash-comment-empty-line-before)",
  },
  Object {
    "column": 18,
    "line": 81,
    "rule": "scss/double-slash-comment-inline",
    "severity": "error",
    "text": "Unexpected inline //-comment (scss/double-slash-comment-inline)",
  },
  Object {
    "column": 18,
    "line": 84,
    "rule": "scss/double-slash-comment-whitespace-inside",
    "severity": "error",
    "text": "Expected a space after // (scss/double-slash-comment-whitespace-inside)",
  },
  Object {
    "column": 1,
    "line": 90,
    "rule": "scss/no-duplicate-dollar-variables",
    "severity": "error",
    "text": "Unexpected duplicate dollar variable $i (scss/no-duplicate-dollar-variables)",
  },
  Object {
    "column": 1,
    "line": 97,
    "rule": "scss/no-duplicate-mixins",
    "severity": "error",
    "text": "Unexpected duplicate mixin font-size-default (scss/no-duplicate-mixins)",
  },
  Object {
    "column": 18,
    "line": 102,
    "rule": "scss/operator-no-newline-after",
    "severity": "error",
    "text": "Unexpected newline after \\"+\\" (scss/operator-no-newline-after)",
  },
  Object {
    "column": 3,
    "line": 107,
    "rule": "scss/operator-no-newline-before",
    "severity": "error",
    "text": "Unexpected newline before \\"+\\" (scss/operator-no-newline-before)",
  },
  Object {
    "column": 15,
    "line": 110,
    "rule": "scss/operator-no-unspaced",
    "severity": "error",
    "text": "Expected single space before \\"+\\" (scss/operator-no-unspaced)",
  },
  Object {
    "column": 15,
    "line": 110,
    "rule": "scss/operator-no-unspaced",
    "severity": "error",
    "text": "Expected single space after \\"+\\" (scss/operator-no-unspaced)",
  },
  Object {
    "column": 3,
    "line": 114,
    "rule": "scss/selector-no-redundant-nesting-selector",
    "severity": "error",
    "text": "Unnecessary nesting selector (&) (scss/selector-no-redundant-nesting-selector)",
  },
  Object {
    "column": 1,
    "line": 45,
    "rule": "at-rule-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown at-rule \\"@unknown\\" (at-rule-no-unknown)",
  },
  Object {
    "column": 3,
    "line": 103,
    "rule": "block-closing-brace-newline-before",
    "severity": "error",
    "text": "Expected newline before \\"}\\" of a multi-line block (block-closing-brace-newline-before)",
  },
  Object {
    "column": 7,
    "line": 107,
    "rule": "block-closing-brace-newline-before",
    "severity": "error",
    "text": "Expected newline before \\"}\\" of a multi-line block (block-closing-brace-newline-before)",
  },
  Object {
    "column": 7,
    "line": 102,
    "rule": "block-opening-brace-newline-after",
    "severity": "error",
    "text": "Expected newline after \\"{\\" of a multi-line block (block-opening-brace-newline-after)",
  },
  Object {
    "column": 4,
    "line": 106,
    "rule": "block-opening-brace-newline-after",
    "severity": "error",
    "text": "Expected newline after \\"{\\" of a multi-line block (block-opening-brace-newline-after)",
  },
  Object {
    "column": 36,
    "line": 52,
    "rule": "declaration-block-semicolon-newline-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\";\\" in a multi-line declaration block (declaration-block-semicolon-newline-before)",
  },
  Object {
    "column": 36,
    "line": 52,
    "rule": "declaration-block-semicolon-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\";\\" (declaration-block-semicolon-space-before)",
  },
  Object {
    "column": 21,
    "line": 56,
    "rule": "declaration-block-trailing-semicolon",
    "severity": "error",
    "text": "Expected a trailing semicolon (declaration-block-trailing-semicolon)",
  },
  Object {
    "column": 13,
    "line": 102,
    "rule": "declaration-colon-newline-after",
    "severity": "error",
    "text": "Expected newline after \\":\\" with a multi-line declaration (declaration-colon-newline-after)",
  },
  Object {
    "column": 10,
    "line": 106,
    "rule": "declaration-colon-newline-after",
    "severity": "error",
    "text": "Expected newline after \\":\\" with a multi-line declaration (declaration-colon-newline-after)",
  },
  Object {
    "column": 3,
    "line": 56,
    "rule": "declaration-empty-line-before",
    "severity": "error",
    "text": "Unexpected empty line before declaration (declaration-empty-line-before)",
  },
  Object {
    "column": 1,
    "line": 1,
    "rule": "no-empty-first-line",
    "severity": "error",
    "text": "Unexpected empty line (no-empty-first-line)",
  },
  Object {
    "column": 5,
    "line": 57,
    "rule": "property-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown property \\"repeat\\" (property-no-unknown)",
  },
  Object {
    "column": 13,
    "line": 110,
    "rule": "unit-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown unit \\"+1\\" (unit-no-unknown)",
  },
  Object {
    "column": 1,
    "line": 103,
    "rule": "indentation",
    "severity": "error",
    "text": "Expected indentation of 4 spaces (indentation)",
  },
  Object {
    "column": 3,
    "line": 107,
    "rule": "indentation",
    "severity": "error",
    "text": "Expected indentation of 4 spaces (indentation)",
  },
]
`;
